### PR TITLE
Remove transitions from docs to comply with warnings from `sphinx-check`

### LINKED ISSUE #490: "Document or section may not begin with a transition" warnings from sphinx-check
## 🐛 Bug

tmp

### PR DESCRIPTION
**Pull request by akihironitta**
## What does this PR do?
Fixes #490.

This PR removes transitions in the docs to resolve warnings:
```
Document or section may not begin with a transition.
```
from CI `sphinx-check`.

---

Below is the example of the changes in the docs. https://pytorch-lightning-bolts.readthedocs.io/en/latest/self_supervised_models.html#byol
### Before the change
tmp

### After the change
tmp

## Before submitting
- [ ] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [ ] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning-bolts/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [ ] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [ ] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests? [not needed for typos/docs]
- [ ] Did you verify new and existing tests pass locally with your changes?
- [ ] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning-bolts/blob/master/CHANGELOG.md)?

## PR review
 - [ ] Is this pull request ready for review? (if not, please submit in draft mode)

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Comment by akihironitta** [comment]
It seems there is no way to disable `Unchanged files with check annotations (Beta)` as the issue https://github.com/actions/toolkit/issues/457 is still open, so I'll try to resolve those warnings from sphinx-check in this PR instead of turning off the feature `Unchanged files with check annotations (Beta)`.

**Comment by Borda** [comment]
yeah, I was thinking rather fix the warnings then suppress them from PRs :]

File: docs/source/losses.rst [hunk 1]
```diff
@@ -39,16 +39,12 @@ DQN Loss
 .. autofunction:: pl_bolts.losses.rl.dqn_loss
     :noindex:
 
----------------
```
Comment (Borda) [inline docs/source/losses.rst:42]: I think that @williamFalcon  made it on purpose, so is there another way how to have a horizontal line?

Comment (akihironitta) [reply docs/source/losses.rst:42]: I looked through the docs, but I'm still not sure if there's another way to draw a horizontal line...
- [reStructuredText Markup Specification](https://docutils.sourceforge.io/docs/ref/rst/restructuredtext.html)
- [Sphinx documentation contents](https://www.sphinx-doc.org/en/master/contents.html)

Comment (Borda) [reply docs/source/losses.rst:42]: could it be this one? https://docutils.sourceforge.io/docs/ref/doctree.html#transition
so let's just remove it at this moment :]

**Comment by Borda** [comment]
mind check the two failing test for make and test docs seem like we need to mock the packages...

**Comment by akihironitta** [comment]
@Borda Would you mind checking the changes again?

**Review by Borda — APPROVED** [review]

**Pull request merged**